include/api/CAnomalyJobConfig.h

```diff
@@ -68,7 +67,7 @@ class API_EXPORT CAnomalyJobConfig {
             std::string m_PartitionFieldName{};
             std::string m_ExcludeFrequent{};
             std::string m_DetectorDescription{};
-            CDetectionRulesJsonParser::TDetectionRuleVec m_CustomRules{};
+            std::size_t m_DetectorIndex{};
```

Comment:
Although probably not the best choice originally, detector index is an `int` everywhere else, so it's best if it's an `int` here too.  (I think originally it was some generic "identifier" instead of an array index, and -1 indicated "unknown".  That's so far in the past that -1 never exists now, but it's still probably best to keep the types consistent across the codebase.)